File: case2/test/page_object.py
from selenium.webdriver.support.ui import WebDriverWait
from selenium.webdriver.common.alert import Alert
from selenium.webdriver.common.by import By

class FishesList(object):

    def __init__(self, driver, target_origin):
        self._driver = driver
        self._target_origin = target_origin

    def open(self):
        self._driver.get(self._target_origin)
        return self

    def wait_for_row_to_finish_loading(self):
        WebDriverWait(self._driver, 5).until(
            lambda d: len(self.get_rows()) != 0
        )
        return self

    def click_select(self):
        self._driver.find_element_by_id('select-button').click()
        return self._driver.switch_to.alert

    def click_all_check(self):
        self._driver.find_element_by_id('all-check').click()
        return self

    def get_rows(self):
        rows = self._driver.find_element_by_id('fishes-list')\
                .find_elements_by_tag_name('tr')
        return list(map(lambda el: FishesListRow(self._driver, el), rows))


class FishesListRow(object):

    def __init__(self, driver, el):
        self._driver = driver
        self._el = el

    def get_name(self):
        return self._el.find_elements_by_tag_name('td')[1].text

    def click_checkbox(self):
        self._el.find_element_by_class_name('select-row').click()
        return self

File: case1/test/spec/PageObject.js
const {By, Key, until} = require('selenium-webdriver');

export class FishList {

  constructor(driver) {
    this.driver = driver;
  }

  async open() {
    await this.driver.get('http://localhost:8080');
    return this;
  }

  async waitForRowToFinishLoading() {
    await this.driver.wait(async () => {
      const rows = await this.getRows();
      return rows.length !== 0;
    }, 5000);
    return this;
  }

  async clickSelect() {
    await this.driver.findElement(By.id('select-button')).click();
    return this.driver.switchTo().alert();
  }

  async clickAllCheck() {
    const check = await this.driver.findElement(By.id('all-check'));
    check.click();
    return this;
  }

  async getRows() {
    const rows = await this.driver.findElement(By.id('fish-list')).findElements(By.tagName('tr'));
    return rows.map(r => {
      return new FishListRow(this.driver, r);
    });
  }
}

export class FishListRow {

  constructor(driver, el) {
    this._driver = driver;
    this._el = el;
  }

  async getName() {
    const cols = await this._el.findElements(By.tagName('td'));
    return cols[1].getText();
  }

  async clickCheckBox() {
    await this._el.findElement(By.className('select-row')).click();
    return this;
  }

}
